clang/lib/AST/ASTStructuralEquivalence.cpp

```diff
@@ -645,6 +645,9 @@ static bool IsStructurallyEquivalent(StructuralEquivalenceContext &Context,
      // It is sufficient to check value of getAsTemplateDecl.
      break;
 
+   case TemplateName::DeducedTemplate:
+     // FIXME: We can't reach here.
+     llvm_unreachable("unimplemented");
```

Comment:
is it unreachable or just unimplemented?
If deduced arguments are only used during partial ordering they should never need to be compared for structural equivalence... maybe. I am not actually sure, but the fix me and the strings should agree with each other!

Reply:
Well it's both, ideally we should implement this, but we don't implement other stuff that would even allow us to reach here.

Reply:
Hum, fair enough. It's hard to me to understand how we could ever get there though